### PR TITLE
Master report fp

### PR DESCRIPTION
**Pull request by jem-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/payment_transfer/models/payment.py [hunk 1]
```diff
@@ -18,35 +18,6 @@ class TransferPaymentAcquirer(models.Model):
     def transfer_get_form_action_url(self):
         return '/payment/transfer/feedback'
 
-    def _format_transfer_data(self):
```
Comment (tivisse) [inline addons/payment_transfer/models/payment.py]: Why this change ? Is it really useful or something to avoid a bug ?

File: addons/report/models/report.py [hunk 2]
```diff
@@ -266,12 +267,27 @@ def get_pdf(self, docids, report_name, html=None, data=None):
...
 
     @api.noguess
-    def get_action(self, docids, report_name, data=None):
+    def get_action(self, docids, report_name, data=None, config=True):
         """Return an action of type ir.actions.report.xml.
 
         :param docids: id/ids/browserecord of the records to print (if not used, pass an empty list)
         :param report_name: Name of the template to generate an action for
         """
+        if ((not self.env.user.company_id.default_report_template) or (not self.env.user.company_id.logo)) and config:
```
Comment (tivisse) [inline addons/report/models/report.py]: Seems louche.

File: addons/payment_transfer/models/payment.py (hunk 1 above)
Comment (fpodoo) [inline addons/payment_transfer/models/payment.py]: We removed the feature "Display on Footer" on bank accounts because it's easier to directly configure your footer with plain text: 1/ easier to find the option rather than setting bank journals, and 2/ the way to display bank data are very country specific. (so the way Odoo renders it is wrong 50% of the time: IBAN/BIC/Swift, RIB, bank address of not, ...)

The same logic applies to default value of wire transfer (which was the only other feature that was using the "Show on Footer" option) --> It's better that the user sets its value himself directly instead of providing something wrong.

Note: This code was even worse, the value was created after the create(), so the user did not even saw this default message when creating the document.

More over, by removing this, it allows to remove the "Display on Footer" option on bank journals.

Comment (fpodoo) [inline addons/payment_transfer/models/payment.py]: An improvement could be to do an onchange on the payment type that sets a default value, but it's not clean neither because you could have strange effect if the user sets Wire Transfer, then Credit Card. (the value will be wrong for credit cards processing)

Even if we decide to do something to improve the default value of bank wire transfers, we can view that as a new feature, in a separate branch.

File: addons/payment_transfer/models/payment.py [hunk 3]
```diff
@@ -18,6 +18,35 @@ class TransferPaymentAcquirer(models.Model):
...
 
+    def _format_transfer_data(self):
+        company_id = self.env.user.company_id.id
+        # filter only bank accounts marked as visible
+        journals = self.env['account.journal'].search([('type', '=', 'bank'), ('use_in_payment', '=', True), ('company_id', '=', company_id)])
+        accounts = journals.mapped('bank_account_id').name_get()
+        bank_title = _('Bank Accounts') if len(accounts) > 1 else _('Bank Account')
+        bank_accounts = ''.join(['<ul>'] + ['<li>%s</li>' % name for id, name in accounts] + ['</ul>'])
+        post_msg = '''<div>
+<h3>Please use the following transfer details</h3>
```
Comment (Yenthe666) [inline addons/payment_transfer/models/payment.py]: @fpodoo perhaps it is better to indent this too so it follows the indentation of the function?

**Comment by nhomar** [comment]
On Wed, Jan 4, 2017 at 1:40 AM, Yenthe V.G <notifications@github.com> wrote:

> @fpodoo <https://github.com/fpodoo> perhaps it is better to indent this
> too so it follows the indentation of the function?
>

Indeed the proper way should be a mail.template isn' it?

Nhomar Hernandez
CEO Vauxoo.
Site: http://vauxoo.com
Twitter: @nhomar
Blog: http://nhomar.com
Github User: https://github.com/nhomar
Odoo Gold Partner
Skype: nhomar00 (Envia mail previo no lo superviso siempre).
HangOut: nhomar@vauxoo.com
Móvil Venezuela:
+58 4144110269
Móvil México:
+52 1 4773933942

**Comment by jem-odoo** [comment]
Landed in master for saas-14, bypassing all the process.

https://github.com/odoo/odoo/commit/3d04e44880fb77812e65342ef25bb8ad06094ddc
https://github.com/odoo/odoo/commit/1d6fa1b2fda58928f87d37cb1e59640acffac2d0
https://github.com/odoo/odoo/commit/66638b6838e2a4a10f750005fdd5062315d8389c
https://github.com/odoo/odoo/commit/2d5ff1cb3d237e31263dec51be9a459c7588bcc3
